crates/common/tedge_config_macros/impl/src/lib.rs

```diff
@@ -188,4 +188,18 @@ mod tests {
         })
         .is_err());
     }
+
+    #[test]
+    fn can_contain_hyphen_separated_fields() {
+        generate_configuration(quote! {
+            device: {
+                #[tedge_config(rename = "type")]
+                ty: String,
+
+                #[tedge_config(rename = "hyphen-separated-field", example = "hsf")]
```

Comment:
One thing I noticed is that the panic was not triggered with `#[tedge_config(rename = "hyphen-separated-field")]` and I also had to add `example = hsf` in the macro. Is this intended behaviour?

Reply:
Yes, the generated tests exist to verify the examples are valid, so there is no test if no example is provided, and if there was a second example two tests would be generated.